src/gridfs/download.rs

```diff
@@ -0,0 +1,181 @@
+use tokio::io::AsyncWriteExt;
+use tokio_util::compat::FuturesAsyncWriteCompatExt;
+
+use super::{options::GridFsDownloadByNameOptions, FilesCollectionDocument, GridFsBucket};
+use crate::{
+    bson::{doc, Bson},
+    error::{ErrorKind, Result},
+    options::{FindOneOptions, FindOptions},
+};
+
+impl GridFsBucket {
+    /// Downloads the contents of the stored file specified by `id` and writes
+    /// the contents to the `destination`.
+    pub async fn download_to_tokio_writer<T>(&self, id: Bson, destination: &mut T) -> Result<()>
```

Comment:
We can actually just take in `T` here instead of `&mut T`. See the relevant API guideline: [C-RW-VALUE](https://rust-lang.github.io/api-guidelines/interoperability.html#c-rw-value). Ditto for the other methods